bastion-executor/src/load_balancer.rs

```diff
@@ -47,46 +51,102 @@ impl LoadBalancer {
     }
 }
 
+/// Maximum number of core supported by mordern computers.
+const MAX_CORE: usize = 40;
+
 ///
 /// Holding all statistics related to the run queue
 ///
 /// Contains:
-/// * Global run queue size
 /// * Mean level of processes in the run queues
 /// * SMP queue distributions
-#[derive(Clone, Debug)]
 pub struct Stats {
-    pub(crate) global_run_queue: usize,
-    pub(crate) mean_level: usize,
-    pub(crate) smp_queues: FxHashMap<usize, usize>,
+    smp_load: [AtomicUsize;MAX_CORE],
+    mean_level: AtomicUsize,
 }
 
-unsafe impl Send for Stats {}
-unsafe impl Sync for Stats {}
+impl Stats{
+    /// new returns LockLessStats
+    pub fn new(num_cores: usize)  -> Stats{
+
+        let smp_load: [AtomicUsize; MAX_CORE] = {
+            let mut data: [MaybeUninit<AtomicUsize>; MAX_CORE] = unsafe{
+                MaybeUninit::uninit().assume_init()
+            };
+            let mut i = 0;
+            while i < MAX_CORE{
+                if i < num_cores {
+                    unsafe{std::ptr::write(data[i].as_mut_ptr(),AtomicUsize::new(0));}
+                    i = i+1; 
+                    continue;   
+                }
+                // MAX is for unused slot.
+                unsafe{std::ptr::write(data[i].as_mut_ptr(),AtomicUsize::new(usize::MAX));}
+                i = i+1;
+            }
+            unsafe {
+                std::mem::transmute::<_, [AtomicUsize; MAX_CORE]>(data)
+            }
+        };
+        Stats{
+            smp_load: smp_load,
+            mean_level: AtomicUsize::new(0),
+        }
+    }
+}
+
+unsafe impl Sync for Stats{}
+unsafe impl Send for Stats{}
+
+impl SmpStats for Stats{
+
+    fn store_load(&self, affinity: usize, load: usize){
+        self.smp_load[affinity].store(load, Ordering::SeqCst);
+    }
+
+    fn get_sorted_load(&self) -> Vec<(usize, usize)>{
+        let mut sorted_load = Vec::new();
+
+        for (i, item) in self.smp_load.iter().enumerate(){
+            let load = item.load(Ordering::SeqCst);
+            // load till maximum core.
+            if load == usize::MAX{
+                break;
+            }
+            sorted_load.push((i, load));
+        }
+        sorted_load.sort_by(|x, y| y.1.cmp(&x.1));
+        sorted_load
+    }
+    
+    fn mean(&self) -> usize{
+        self.mean_level.load(Ordering::SeqCst)
+    }
+
+    fn update_mean(&self){
+        let mut sum: usize = 0;
+        
+        for item in self.smp_load.iter(){
+            let load = item.load(Ordering::SeqCst);
+            
+            if load == usize::MAX {
+                break;
+            }
+            sum = sum + load;
```

Comment:
sum.checked_add(load) and break the loop and exit when returns err. since the previous atomic op's result is gonna be messed up.

Reply:
Done